Building cargo 1.72 on Kirkstone fails: could not compile `proc-macro2` (lib)

## Version(s) of meta-rust
`8af044c0edabf1b7e1aac4762af677e058900f92`

## Version(s) of poky and/or oe-core
poky: `f53ab3a2ff206a130cdc843839dd0ea5ec4ad02f`
oe-core: `9e539aaed608a03b1128a7361426703d8f9ba705`

## Expected result
Running `bitbake cargo` builds cargo (and all dependencies)

## Actual result
Build fails with the following error:

```
error: could not compile `proc-macro2` (lib) due to previous error; 2 warnings emitted

Caused by:
  process didn't exit successfully: `rustc --crate-name proc_macro2 --edition=2018 /home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/rustc-1.72.0-src/vendor/proc-macro2-1.0.60/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="proc-macro"' -C metadata=6a2107cbb96f0acc -C extra-filename=-6a2107cbb96f0acc --out-dir /home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/build/target/aarch64-poky-linux/release/deps --target aarch64-poky-linux -C linker=/home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/wrapper/target-rust-ccld -L dependency=/home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/build/target/aarch64-poky-linux/release/deps -L dependency=/home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/build/target/release/deps --extern unicode_ident=/home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/build/target/aarch64-poky-linux/release/deps/libunicode_ident-e239b614f7e8ae95.rmeta --cap-lints allow -L /home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0/recipe-sysroot/usr/lib/rust --remap-path-prefix=/home/user/yocto/build/tmp/work/armv8a-poky-linux/cargo/1.72.0-r0=/usr/src/debug/cargo/1.72.0-r0 --cfg wrap_proc_macro` (exit status: 1)
```

## Steps to reproduce
* Add `meta-rust` to existing kirkstone setup. 
* Set `RUSTVERSION = "1.72%"` in local.conf so it uses the newest version provided by `meta-rust`.

As a workaround I've reverted back to version 1.71 by setting `RUSTVERSION = "1.71%"`, here the build succeeds.

I am experiencing the same issue, though if I try to apply the patch suggested it seems cargo cannot find any external crates used in my project:

```sh
[...]
| error[E0463]: can't find crate for `rand`
|   --> acsctl/src/trunking/trunk/mod.rs:19:5
|    |
| 19 | use rand;
|    |     ^^^^ can't find crate
| 
| error[E0463]: can't find crate for `nix`
|  --> acsctl/src/trunking/tun.rs:9:5
|   |
| 9 | use nix::fcntl::{F_SETFL, OFlag, fcntl};
|   |     ^^^ can't find crate
| 
| error[E0463]: can't find crate for `nix`
|   --> acsctl/src/trunking/tun.rs:10:5
|    |
| 10 | use nix::ioctl_write_int;
|    |     ^^^ can't find crate
| 
[...]
```

Does anyone know how to solve this issue? It seems my recipe is incapable of finding any crates after i applied the patch.
 Is this patch still valid? 
 I am running on version rustc version 1.82.0.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -33,6 +33,7 @@ jobs:
           echo 'TOOLCHAIN_HOST_TASK:append = " packagegroup-rust-cross-canadian-${MACHINE}"' >> conf/local.conf
           echo 'PREFERRED_PROVIDER_virtual/kernel = "linux-dummy"' >> conf/local.conf
           echo 'INHERIT:remove = "uninative"' >> conf/local.conf
+          echo 'require conf/distro/include/rust_versions.inc' >> conf/local.conf
       - name: Run bitbake
         run: |
           cd poky

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -28,7 +28,7 @@ jobs:
           . oe-init-build-env
           mv ../../sstate-cache . || true
           bitbake-layers add-layer ../../meta-openembedded/meta-oe
-          bitbake-layers add-layer ../../meta-rust
+          sed -ie "/^BBLAYERS/a $(readlink -f ../../meta-rust) \\\\" conf/bblayers.conf
           echo 'PARALLEL_MAKE:pn-rust-llvm-native = "-j2"' >> conf/local.conf
           echo 'TOOLCHAIN_HOST_TASK:append = " packagegroup-rust-cross-canadian-${MACHINE}"' >> conf/local.conf
           echo 'PREFERRED_PROVIDER_virtual/kernel = "linux-dummy"' >> conf/local.conf

--- a/classes-recipe
+++ b/classes-recipe
@@ -1,0 +1,1 @@
+classes/

--- a/classes/crate-fetch.bbclass
+++ b/classes/crate-fetch.bbclass
@@ -8,6 +8,9 @@
 #
 
 python () {
+        import sys
+        layerdir = d.getVar("RUSTLAYER")
+        sys.path.insert(0, layerdir + "/lib")
         import crate
         bb.fetch2.methods.append( crate.Crate() )
 }

--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -1,4 +1,5 @@
 # We have a conf and classes directory, append to BBPATH
+RUSTLAYER = "${LAYERDIR}"
 BBPATH .= ":${LAYERDIR}"
 
 # We have a recipes directory, add to BBFILES

--- a/recipes-devtools/rust/libstd-rs_%.bbappend
+++ b/recipes-devtools/rust/libstd-rs_%.bbappend
@@ -1,0 +1,1 @@
+S = "${RUSTSRC}/library/sysroot"

